### DISCUSSION TITLE
Disable "Recent" Section

### DISCUSSION THREAD

**Discussion by cdt-eth** [rainbow-me/rainbowkit#625] [Q&A]
Is there a way to disable the the automatically created "Recent" section from showing? So just keeping my pre-determined group(s) laid out as they always were.

For instance, say I have two sections: `Recommended` & `Hardware` and i use a couple, now there's a 3rd section and the two original sections only have 1 each, and having a group 1 doesn't look great.

fyi: i search for "disable recent" & "recent" before asking this Q
 
<img width="269" alt="Screen Shot 2022-07-29 at 9 54 39 AM" src="https://user-images.githubusercontent.com/26611339/181775432-0e569fc8-01bf-4714-a1e1-7f8d9316e5b3.png">

**Comment by nickbytes** [comment]
I don't think we have an option to disable the Recent section right now, but I like this suggestion. Will add it to our roadmap discussion.

**Reply by cdt-eth** [reply]
sick, appreciate the consideration

**Comment by peduarte** [comment]
i think on the desktop modal experience it can be confusing because the wallets get plucked from their previous wallet list, into a newly created recent list.

on mobile, the experience is a bit different. wallets are into a unique list, so instead of being plucked, the list gets re-ordered. with the most recently used wallets being first on the list. like this:

<img width="412" alt="CleanShot 2022-09-12 at 14 24 04@2x" src="https://user-images.githubusercontent.com/372831/189652567-b6692dd0-1bd2-4e91-9ef3-baf8f623b2bc.png">

i think that especially on mobile, this can be useful for returning users.

---

so now that the differences between desktop and mobile are out of the way, i'd like to propose a couple of potential solutions:

#### solution A: disable the most recent functionality completely
this simply means that the order of wallets will always respect the wallet list configuration. both on desktop and mobile

#### solution B: disable on desktop, keep current mobile implementation
this means that on mobile, since the wallet list "real estate" isnt big enough, we continue to re-order the recently used wallets to the beginning of the list, but we remove the "recent" 

#### solution C: disable "recent group" on desktop, but re-order the wallet list
basically means desktop having the same behaviour as mobile, where most recently used wallets are first in the list

thanks!

**Reply by peduarte** [reply]
chatted with mikey about it. it looks like we're keen of "solution C". which is cool, doesnt even require a prop, its just the default behaviour for all because by removing the implicit "recent group" it probably won't annoy developers. and it still benefit returning users since their most recent wallet is still first in the group

**Reply by peduarte** [reply]
@cdt-eth curious to know what you think as well!

**Reply by cdt-eth** [reply]
@peduarte i'm good with C and the reorder. like you said "can be confusing because the wallets get plucked from their previous wallet list, into a newly created recent list" is my main point. i'd rather configure my list(s) and them stay within the params set. reorder recently used is fine, but adding a 3rd group, when i've denoted 2 groups, is kind of annoying and makes me not what to be opinionated and have specific groups at all.

**Reply by peduarte** [reply]
https://github.com/rainbow-me/rainbowkit/pull/738

**Reply by cdt-eth** [reply]
lfg

**Comment by markdalgleish** [comment]
@peduarte That's a good point that you might want different behaviour across desktop and mobile.

If we add a prop to `RainbowKitProvider` to support disabling the "Recent" section, I wonder if it would be a good idea to support responsive props? e.g.

```tsx
<RainbowKitProvider showRecentWallets={{ mobile: true, desktop: false }}>
```

**Reply by peduarte** [reply]
thanks! i think if we want to go down the path of completely disabling recent wallets, we can definitely consider different behaviour for mobile/desktop. but i think mikey is leaning towards a unified treatment ([basically solution C](https://github.com/rainbow-me/rainbowkit/discussions/625#discussioncomment-3632855))

**Reply by peduarte** [reply]
keen to know what you think
